vllm/inputs/preprocess.py

```diff
@@ -486,6 +487,40 @@ def _build_enc_dec_llm_inputs(
             decoder=decoder_inputs,
         )
 
+    def _handle_multimodal_enc_dec_inputs(
+        self,
+        inputs: SingletonInputs,
+    ) -> Tuple[SingletonInputs, SingletonInputs]:
+        """
+        For encoder/decoder models only:
+        Separate Encoder/Decoder inputs from a MultiModalEncDecInputs
+        """
+        encoder_inputs: SingletonInputs
+        decoder_inputs: SingletonInputs
+        if inputs["type"] == "multimodal":
+            # Multimodal data inputs
+            assert ("encoder_prompt" in inputs
+                    and "encoder_prompt_token_ids" in inputs)
+            inputs = cast(MultiModalEncDecInputs, inputs)
+            encoder_inputs = token_inputs(
+                prompt=inputs["encoder_prompt"],
+                prompt_token_ids=inputs["encoder_prompt_token_ids"],
+            )
+            decoder_inputs = MultiModalInputs(
+                type="multimodal",
+                prompt=inputs["prompt"],
+                prompt_token_ids=inputs["prompt_token_ids"],
+                mm_kwargs=inputs["mm_kwargs"],
+                mm_placeholders=inputs["mm_placeholders"],
+            )
+        elif inputs["type"] == "token":
+            # Text-only inputs
+            encoder_inputs = token_inputs(prompt="", prompt_token_ids=[])
+            decoder_inputs = inputs
+        else:
+            raise AssertionError("This line should be unreachable.")
```

Comment:
```suggestion
            assert_never(inputs)
```

For better type checking